packages/algolia/src/index.ts

```diff
@@ -140,103 +136,111 @@ interface IndexToAlgoliaNextraOptions {
   docsBaseDir: string;
   source: AlgoliaRecordSource;
   domain: string;
+  sitemapXmlPath: string;
   objectsPrefix?: string;
 }
 
-async function getMetaFromFile(path: string) {
-  try {
-    return JSON.parse(await readFile(path, 'utf8'));
-  } catch {
-    /* ignore if _meta.json doesn't exist */
+async function getSitemapUrlsXml(sitemapXmlPath: string): Promise<string[]> {
+  if (!sitemapXmlPath) {
+    throw new Error('Sitemap path not provided');
+  }
+  if (!existsSync(sitemapXmlPath)) {
+    throw new Error(`Sitemap at path "${sitemapXmlPath}" does not exist`);
+  }
+  const sitemap = new XMLParser().parse(await readFile(sitemapXmlPath, 'utf-8'));
+  const sitemapUrls: { loc: string }[] = sitemap?.urlset?.url;
+  if ((sitemapUrls?.length || 0) === 0) {
+    console.log(sitemapXmlPath, JSON.stringify(sitemap, null, '  '));
+    throw new Error('Sitemap urls not found at path urlset>url');
+  }
+  const urls = sitemapUrls.map(({ loc }) => loc).filter(Boolean);
+  if ((urls?.length || 0) === 0) {
+    console.log(sitemapXmlPath, JSON.stringify(sitemap, null, '  '));
```

Comment:
```suggestion
```

Reply:
Intentionally keeping it there to understand how the sitemap looks like when the command errors out.

Reply:
Console.error is better?

Reply:
The log is not an error, its just an info about the contents of the xml. The error is thrown below. We could however change it to console.debug but the subtleties are not that important imho...